### PR TITLE
Fixed #24257 -- Updated i18n's handling of messages with percent signs

### PR DESCRIPTION
**Pull request by beck**
The in-depth explanation of these changes:
https://code.djangoproject.com/ticket/24257#comment:4

### CODE REVIEW

File: tests/i18n/test_percents.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+from django.utils.encoding import force_text
+from django.utils.translation import activate, get_language, trans_real
+
+SAMPLEPROJECT_DIR = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), 'sampleproject')
+SAMPLEPROJECT_LOCALE = os.path.join(SAMPLEPROJECT_DIR, 'locale')
+
+
+@override_settings(LOCALE_PATHS=[SAMPLEPROJECT_LOCALE])
+class SampleprojectTests(TestCase):
```
Comment (charettes) [inline tests/i18n/test_percents.py]: Use `SimpleTestCase`

**Comment by timgraham** [comment]
@claudep, could you take a look at this and say if the general approach seems okay? If so, I'll be happy to review it in more detail.

**Comment by claudep** [comment]
This looks like a solid refactoring. We can indeed enter detail review. As for me, Doug, could you rename `sampleproject` into something more explicit (`percentproject`, ...)?

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by beck** [comment]
> rename sampleproject into something more explicit

My motivation for using a generic name was for reuse.

I found it immensely helpful when testing i18n to use output/translation catalogs generated by Django (instead of sample data assumed to be the same as what django would output). The po files are also easy to update if/when extraction or template language change.

But maybe I'm wrong here?  And the sampleproject should not be reused outside of this pull?

**Comment by beck** [comment]
@timgraham thanks for the review, I'll tend to it all ASAP (prob this weekend)

**Comment by beck** [comment]
I rebased off of master. The github comments on the original commits are no longer seen on this pull but can be found here: https://github.com/beck/django/pull/2

@timgraham thanks for the review and I believe addressed each comment.

I tried my best to reword comments in code and commits to be intelligible all, with or without familiarity with i18n.

**Comment by timgraham** [comment]
Is it possible to allow the tests to pass after the "refactor" commit (or should all 3 commits be squashed)?

**Comment by beck** [comment]
if each commit needs to pass, then squashed is probably best.

**Comment by timgraham** [comment]
merged in b7508896fbe19ec2cdeb81565cd587091b6b68d0, thanks!

**Comment by beck** [comment]
Woo.  @timgraham thank you for all the help

**Pull request closed without merging**